### DISCUSSION TITLE
Has anyone seen an issue installing with Spack

### DISCUSSION THREAD

**Discussion by EvanDodd** [BLAST-WarpX/warpx#5303] [Installation]
I was trying to install WarpX with Spack, and during the installation of WarpX it stopped with the message:

 1781 make: *** No rule to make target 'pip_wheel'. Stop.

There was more output, and I've edited for brevity. Also, it looks like all of the dependencies were installed properly, so this looks to be from the WarpX portion. This is a Cray machine. Thanks,

Evan Dodd

**Comment by ax3l** [comment]
Hi @EvanDodd,

Which version of Spack are you using? We recently did a major overhaul of `warpx` in the latest `develop` branch of Spack.

Can you please zip and upload the whole output and if you did any specific configurations (e.g., external packages reusing Cray modules) then those files here, too?

What exact command and Spack specs did you use to build?

The `pip_wheel` target in WarpX is only built if Python bindings are requested. I took a look again at the latest file and this looks good, so we need to debug what goes wrong on this Cray machine together.

Best,
Axel

**Reply by EvanDodd** [reply]
@ax3l,

I had forgotten that all of Los Alamos' open HPC is down today. I can't find a path to the file system, much less get that actual version of Spack. I apologize, I should be aware of this but was on travel last week. The machines are supposed to return 5 pm tomorrow.

Since Spack didn't finish, what output there was sat in /tmp. I assume this is what you mean, but let me know if I'm wrong.

I initially tried: spack install py-warpx. spack info says that py-warpx is depricated, so it tried: spack install warpx. My preference is for running through python scripts, if that isn't too slow. However, other that initial attempt at py-warpx, I never expicitly told Spack to use python.

Thank you,
Evan

**Reply by ax3l** [reply]
@EvanDodd no worries, we can continue when the machine is back on.

Yes, the new spec to use is `warpx +python` since 24.08.

**Reply by EvanDodd** [reply]
@ax3l, Rocinante is back. The Spack version is: 0.23.0.dev0 (3589edcc6d838009aba97cfe9d66e34d1e579715)

When I went to zip up the directory of output that you requested, it ended up 70 MB after the zip. A number of .so files were actually created. Instead, I have of uploaded a number of the text files that were output since they easily fit the file size limits.

If there is a better way to transfer the files, or a better subset of files to transfer, then please let me know. Do you or anyone you work with have access to the HPC machines here at LANL? Thanks,

Evan


[spack-build-01-cmake-out.txt](https://github.com/user-attachments/files/17139930/spack-build-01-cmake-out.txt)
[spack-build-02-build-out.txt](https://github.com/user-attachments/files/17139931/spack-build-02-build-out.txt)
[spack-build-env-mods.txt](https://github.com/user-attachments/files/17139932/spack-build-env-mods.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/17139933/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/17139934/spack-build-out.txt)
[spack-configure-args.txt](https://github.com/user-attachments/files/17139935/spack-configure-args.txt)
